velox/dwio/parquet/reader/StructColumnReader.cpp

```diff
@@ -22,18 +22,47 @@ namespace facebook::velox::parquet {
 StructColumnReader::StructColumnReader(
     const std::shared_ptr<const dwio::common::TypeWithId>& dataType,
     ParquetParams& params,
-    common::ScanSpec& scanSpec)
+    common::ScanSpec& scanSpec,
+    const RowTypePtr& outputType,
+    memory::MemoryPool& pool)
     : SelectiveStructColumnReader(dataType, dataType, params, scanSpec) {
   auto& childSpecs = scanSpec_->stableChildren();
+  std::vector<int> missingFields;
   for (auto i = 0; i < childSpecs.size(); ++i) {
     if (childSpecs[i]->isConstant()) {
       continue;
     }
-    auto childDataType = fileType_->childByName(childSpecs[i]->fieldName());
+    const auto& fieldName = childSpecs[i]->fieldName();
+    if (outputType && !fileType_->containsChild(fieldName)) {
```

Comment:
We need to decide what is the schema evolution strategy we want here.  In our data warehouse, columns are not matched by name but by position, so any extra fields added need to be at the end of the children list.  This allows column renaming.  If we match by name here, we will lose the renaming functionality and this seems quite important in most data warehouse.

Reply:
Thanks for your comment. Does that mean for a `row(a, c)` struct schema in parquet, the expected output can only be like `row(a, c, xxx, ...)`? In Spark, there is no such limitation to extra child fields.

Reply:
Yes new subfields can only be appended.  So in plain vanilla Spark, field renaming is not supported?  There is also a third way to match by field ID (e.g. Iceberg), we need to start draft some design about this to cover all three cases.

Reply:
How does field renaming is conducted in the data warehouse you mentioned? In Spark, for query like `select a as b`, it adds a projection node with `Alias` expression after scan.
And what do you suggest for the design, should I added some notes in this PR or somewhere else?

Reply:
With matching by name you need to know all the old field names (`a` in your query) in all old files, which is not practical in a normal data warehouse.  I would suggest we pause this PR for a bit and design the right way to allow matching columns in different ways first.

Reply:
Thanks. That looks good to me. Convert this PR to draft for now.